### DISCUSSION TITLE
parameter limit&filter using at the same time

### DISCUSSION THREAD

**Discussion by honue** [DIYgod/RSSHub#13674] [Q&A] [answered]
When I use the two parameters simultaneously, I noticed that the program first applies the limit and then the filter, filtering within the sub-items. This causes a problem where there are enough items, but they cannot be displayed. This order seems illogical. Should we adjust the execution sequence?

**Comment by honue** [comment]
1 1 1 1 2 2 2 1 1 1 1, limit=5 filter=1, ret is ['1,1,1,1']

**Comment by TonyRL** [comment]
The order of processing common parameters is [here](https://github.com/DIYgod/RSSHub/blob/0acae72394c4f34ffa148d891c8da9082b23889b/lib/middleware/parameter.js#L151). They're being processed from top to bottom.

**Reply by honue** [reply]
I know that's the case. What I mean is that the project team has considered putting the limit after the filter？

**Reply by honue** [reply]
In other words, what were the considerations behind placing limit processing before filter?

**Reply by TonyRL** [reply, marked as answer]
The order is changed in a refactor(#3200). Filtering does come before limit before #3200 https://github.com/DIYgod/RSSHub/blame/a2600c4c7933fcc6a610a25458a839af48f020dd/lib/middleware/parameter.js. Feel free to submit a PR if you want to.

**Reply by honue** [reply]
I'm very grateful for your reply.

**Comment by honue** [comment]
#13682
